packages/generator-rest/src/request-builder/operation-request-builder-class.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  ClassDeclarationStructure,
+  MethodDeclarationStructure,
+  OptionalKind,
+  ParameterDeclarationStructure,
+  Scope,
+  StructureKind
+} from 'ts-morph';
+import { toPascalCase, toPropertyFormat } from '@sap-cloud-sdk/core';
+import {
+  OpenApiOperation,
+  OpenApiPath,
+  OpenApiServiceMetadata
+} from '../open-api-types';
+import {
+  pathParamToParamStructure,
+  refNameToParamStructure
+} from './api-request-builder-class';
+
+export function operationRequestBuilderClass(
+  metadata: OpenApiServiceMetadata,
+  openApiPath: OpenApiPath,
+  operation: OpenApiOperation
+): ClassDeclarationStructure {
+  return {
+    kind: StructureKind.Class,
+    name: `${toPascalCase(operation.operationName)}RequestBuilder`,
+    methods: method(metadata, openApiPath, operation),
+    extends: 'RestRequestBuilder',
+    isExported: true,
+    ctors: [
+      {
+        parameters: [
+          ...addPublicScope([getRequestConfigParameter()]),
+          ...addPublicScope(
+            pathParamToParamStructure(openApiPath.pathParameters)
+          ),
+          ...addPublicScope(
+            refNameToParamStructure(operation.requestBodySchemaRefName)
+          )
+        ],
+        statements: 'super(requestConfig);'
+      }
+    ]
+  };
+}
+
+function addPublicScope(
+  objs: OptionalKind<ParameterDeclarationStructure>[]
+): OptionalKind<ParameterDeclarationStructure>[] {
+  return objs.map(o => ({ ...o, scope: Scope.Public }));
+}
+
+function method(
+  metadata: OpenApiServiceMetadata,
+  openApiPath: OpenApiPath,
+  operation: OpenApiOperation
+): MethodDeclarationStructure[] {
+  return [
+    {
+      kind: StructureKind.Method,
+      name: 'execute',
+      isAsync: true,
```

Comment:
here the return type must be included. But I guess it is not see easy...

Reply:
Without the explicit return type, it is `Promise<AxiosResponse<PriceCalculateResponse>>` now.
The reason why it's not so easy is because you have to know which response code you refer to.
In this case,  the it looks like:
```
200: {
  content:{
    application-json:{
      schema: {
        $ref: "#/components/schemas/PriceCalculateResponse"
      }
    }
  }
}
400:{}
default:{} 
```
You can argue, just use the one with the 2XX pattern.
We can also try to check the number of 2XX, 
- if there is one occurrence, we use it as return type
- else, we leave it empty

Reply:
tested with the VS Code, the `execute()` function returns the type `Promise<AxiosResponse<PriceCalculateResponse>>`.
I will add the return type with the strategy mentioned above so the potential compilation errors are detected earlier when breaking changes are introduced.